File: src/tomek.js/compiler/template/stencil_node.rb
# To change this template, choose Tools | Templates
# and open the template in the editor.

class StencilNode < TemplateNode
	
	@property_name
	@placeholder_node
	
	def property_name
		@property_name
	end
	
	def initialize node
    super
		@property_name = node.name.to_s
		@placeholder_node = ComponentNode.new "TContent"
		@placeholder_node.parent= self
		@children.push @placeholder_node
  end
	
	def add_child c
		@placeholder_node.add_child c
	end
	
end

File: src/tomek.js/compiler/template/StencilNode.ts
/// <reference path="../typings/libxmljs/libxmljs.d.ts" />

import libxmljs  = require('libxmljs');
import TemplateNode from './TemplateNode';
import ComponentNode from './ComponentNode';

export default class StencilNode extends TemplateNode {

	propertyName: string;
	placeholderNode: TemplateNode;

	constructor( xmlNode: libxmljs.Element ){
		super();
		this.propertyName = xmlNode.name();
		this.placeholderNode = new ComponentNode("TContent");
		this.placeholderNode.parent = this;
		this.children.push( this.placeholderNode );
	}

	addChild(node: TemplateNode) {
		this.placeholderNode.addChild(node);
	}

}